rerun_cpp/src/rerun/indicator_component.hpp

```diff
@@ -37,9 +40,13 @@ namespace rerun {
 
         /// Creates an arrow ComponentBatch from an array of IndicatorComponent components.
         static Result<std::shared_ptr<arrow::Array>> to_arrow(
-            const components::IndicatorComponent<Name_>*, size_t
+            const components::IndicatorComponent<Name_>*, size_t num_instances
         ) {
-            return components::indicator_arrow_array();
+            if (num_instances == 1) {
+                return components::indicator_arrow_array();
+            } else {
+                return components::indicator_arrow_array(num_instances);
+            }
```

Comment:
Why not this?

```suggestion
            return components::indicator_arrow_array(num_instances);
```

Reply:
one of them is optimized to not do a an allocation every time (see different function decls). Will add a comment!